relayer/relays/beacon/main.go

```diff
@@ -110,15 +110,34 @@ func (r *Relay) Sync(ctx context.Context) error {
 		}
 	}
 
-	logrus.Info("done with sync committee updates ")
+	lastFinalizedHeader, err := r.writer.getLastStoredFinalizedHeader()
+	if err != nil {
+		logrus.WithError(err).Error("unable to get last finalized header")
 
-	logrus.Info("starting to sync finalized headers")
+		return err
+	}
 
-	_, _, err = r.SyncFinalizedHeader(ctx)
+	lastFinalizedSlot, err := r.writer.getLastStoredFinalizedHeaderSlot()
 	if err != nil {
+		logrus.WithError(err).Error("unable to get last finalized header slot")
+
 		return err
 	}
 
+	r.syncer.Cache.FinalizedHeaders = append(r.syncer.Cache.FinalizedHeaders, lastFinalizedHeader)
+
+	logrus.WithFields(logrus.Fields{
+		"hash": lastFinalizedHeader,
+		"slot": lastFinalizedSlot,
+	}).Info("set cache: last finalized header")
+
+	logrus.Info("starting to sync finalized headers")
+
+	err = r.SyncHeaders(ctx)
```

Comment:
I see we end up calling SyncHeaders several times in this method. Here, and then in a loop further below on line 131.

Is this expected? Shouldn't we only call it in the loop?

Reply:
I called `SyncHeaders` here too because the ticker started with a delay. I wanted to immediately call `SyncHeaders` (and not wait for the ticker delay) and then call it in the ticker increments. I found a way to call the function immediately, changes in https://github.com/Snowfork/snowbridge/pull/668/commits/27119fdd6815bfb60e1684778a3347fa3d6a1582.

Referenced: https://stackoverflow.com/questions/32705582/how-to-get-time-tick-to-tick-immediately